src/types.ts

```diff
@@ -155,6 +155,17 @@ export type PrincipalStats = {
   group: number;
 };
 
+/**
+ * Paginated Result
+ */
+export type PaginatedResult<T> = {
+  principals: T[];
+  total: number;
+  page: number;
+  pageSize: number;
+  hasNextPage: boolean;
+}
```

Comment:
Given the name of this type, I think this can use a small tweak to be useful for other lookups that aren't principals:

```suggestion
export type PaginatedResult<T> = {
  items: T[];
  total: number;
  page: number;
  pageSize: number;
  hasNextPage: boolean;
}
```